dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ClientRateLimitHelper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.collections.map.CaseInsensitiveMap;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helps a Java Client making http requests stay within the server's rate limits.
+ *
+ * Defaults to honoring the header fields from https://datatracker.ietf.org/doc/draft-ietf-httpapi-ratelimit-headers/. Note that GitHub
+ * precedes the header names with "x-", a common practice for non-standard headers (the link is a draft, not formally accepted).
+ *
+ * <ul>
+ *     <li>RateLimit-Limit</li>
+ *     <li>RateLimit-Remaining</li>
+ *     <li>RateLimit-Reset</li>
```

Comment:
I navigated to the mentioned spec, and it proposes a different scheme (a single `RateLimit` header) and mentions the drawbacks to using the headers above:
https://www.ietf.org/archive/id/draft-ietf-httpapi-ratelimit-headers-08.html#name-ratelimit-header-fields-cur
I got confused because a different proposal said that the value of the `X-RateLimit-Reset` field was a time until reset in seconds, rather than epoch seconds:
https://www.ietf.org/archive/id/draft-polli-ratelimit-headers-02.html#name-ratelimit-reset
Both github and zenodo appear to support the above headers, and the value of the "Reset" header is epoch seconds. 
 Alas, the single `RateLimit` header doesn't appear to be supported by github or zenodo.